packages/by-name/kata/kata-runtime/0023-misc-use-compressed-policy.patch

```diff
@@ -0,0 +1,223 @@
++	Policy []byte `protobuf:"bytes,1,opt,name=policy,proto3" json:"policy,omitempty"`
+ }
+ 
+ func (x *SetPolicyRequest) Reset() {
+@@ -4213,11 +4213,11 @@ func (*SetPolicyRequest) Descriptor() ([]byte, []int) {
+ 	return file_agent_proto_rawDescGZIP(), []int{67}
+ }
+ 
+-func (x *SetPolicyRequest) GetPolicy() string {
++func (x *SetPolicyRequest) GetPolicy() []byte {
+ 	if x != nil {
+ 		return x.Policy
+ 	}
+-	return ""
++	return nil
+ }
+ 
+ type MemAgentMemcgConfig struct {
+@@ -4904,7 +4904,7 @@ var file_agent_proto_rawDesc = []byte{
+ 	0x68, 0x12, 0x12, 0x0a, 0x04, 0x73, 0x69, 0x7a, 0x65, 0x18, 0x02, 0x20, 0x01, 0x28, 0x04, 0x52,
+ 	0x04, 0x73, 0x69, 0x7a, 0x65, 0x22, 0x2a, 0x0a, 0x10, 0x53, 0x65, 0x74, 0x50, 0x6f, 0x6c, 0x69,
+ 	0x63, 0x79, 0x52, 0x65, 0x71, 0x75, 0x65, 0x73, 0x74, 0x12, 0x16, 0x0a, 0x06, 0x70, 0x6f, 0x6c,
+-	0x69, 0x63, 0x79, 0x18, 0x01, 0x20, 0x01, 0x28, 0x09, 0x52, 0x06, 0x70, 0x6f, 0x6c, 0x69, 0x63,
++	0x69, 0x63, 0x79, 0x18, 0x01, 0x20, 0x01, 0x28, 0x0c, 0x52, 0x06, 0x70, 0x6f, 0x6c, 0x69, 0x63,
+ 	0x79, 0x22, 0xfc, 0x03, 0x0a, 0x13, 0x4d, 0x65, 0x6d, 0x41, 0x67, 0x65, 0x6e, 0x74, 0x4d, 0x65,
+ 	0x6d, 0x63, 0x67, 0x43, 0x6f, 0x6e, 0x66, 0x69, 0x67, 0x12, 0x1f, 0x0a, 0x08, 0x64, 0x69, 0x73,
+ 	0x61, 0x62, 0x6c, 0x65, 0x64, 0x18, 0x01, 0x20, 0x01, 0x28, 0x08, 0x48, 0x00, 0x52, 0x08, 0x64,
+diff --git a/src/tools/genpolicy/Cargo.lock b/src/tools/genpolicy/Cargo.lock
+index 96faae9ad7c8ab2a11b5f2eebbdd648423c82a79..1979ea34cda0f6ef860b1d242751bef8ad37dcb4 100644
+--- a/src/tools/genpolicy/Cargo.lock
++++ b/src/tools/genpolicy/Cargo.lock
+@@ -75,6 +75,19 @@ dependencies = [
+  "wait-timeout",
+ ]
+ 
++[[package]]
++name = "async-compression"
++version = "0.4.27"
++source = "registry+https://github.com/rust-lang/crates.io-index"
++checksum = "ddb939d66e4ae03cee6091612804ba446b12878410cfa17f785f4dd67d4014e8"
++dependencies = [
++ "flate2",
++ "futures-core",
++ "memchr",
++ "pin-project-lite",
++ "tokio",
++]
++
+ [[package]]
+ name = "async-trait"
+ version = "0.1.68"
+@@ -1289,6 +1302,7 @@ name = "kata-agent-policy"
+ version = "0.1.0"
+ dependencies = [
+  "anyhow",
++ "async-compression",
+  "json-patch",
+  "nix 0.24.3",
+  "regorus",
+diff --git a/src/tools/genpolicy/src/policy.rs b/src/tools/genpolicy/src/policy.rs
+index 10de9e825bb238e37da210a2b73ccfd62e20394c..e7b55eef4232a95785fbfdc466bc444f5f904ff5 100644
+--- a/src/tools/genpolicy/src/policy.rs
++++ b/src/tools/genpolicy/src/policy.rs
+@@ -578,7 +578,14 @@ impl AgentPolicy {
+         if self.config.raw_out {
+             std::io::stdout().write_all(policy.as_bytes()).unwrap();
+         }
+-        general_purpose::STANDARD.encode(policy.as_bytes())
++
++        let mut compressed_policy = Vec::new();
++        let mut gz = flate2::GzBuilder::new()
++            .write(&mut compressed_policy, flate2::Compression::best());
++        gz.write_all(policy.as_bytes()).unwrap();
++        gz.finish().unwrap();
++        
```

Comment:
```suggestion
+
```

`check_patches` does not seem to care, but `git am` does 🤷🏼‍♀️

Reply:
It does, try it out with `git am`, git complains 🤷🏼‍♀️

Reply:
```
[...]patch:213: trailing whitespace.
        
warning: 1 line adds whitespace errors.
```
This one? You can actually configure this: https://stackoverflow.com/a/12396793. I probably did not notice because I'm using `-3` (maybe I shouldn't be using it).